proposals/Simple-programs.md

```diff
@@ -0,0 +1,169 @@
+
+``` c#
+// File 1 - any statements
+if (args.Length == 0
+    || !int.TryParse(args[0], out int n)
+    || n < 0) return;
+Console.WriteLine(Fib(n).curr);
+
+// File 2 - only local functions
+(int curr, int prev) Fib(int i)
+{
+    if (i == 0) return (1, 0);
+    var (curr, prev) = Fib(i - 1);
+    return (curr + prev, curr);
+}
+```
+
+### Semantics
+
+If any top-level statements are present in any compilation unit of the program, the meaning is as if
+they were combined in the block body of a `Main` method of a `Program` class in the global namespace,
+as follows:
+
+``` c#
+static class Program
+{
+    static async Task Main()
+    {
+        // File 1 statements
+        // File 2 local functions
+        // ...
+    }
+}
+```
+
+Note that the names "Program" and "Main" are used only for illustrations purposes, actual names used by
+compiler are implementation dependent and neither the type, nor the method can be referenced by name from
+source code.
+
+The method is designated as the entry point of the program. Explicitly declared methods that by convention 
+could be considered as an entry point candidates are ignored. A warning is reported when that happens. It is
+an error to specify `-main:<type>` compiler switch.
+
+If any one compilation unit has statements other than local function declarations, statements from that
+compilation unit occur first. This causes it to be legal for local functions in one file to reference
+local variables in another. The order of statement contributions (which would all be local functions)
+from other compilation units is undefined.
+
+If `await` expressions and other async operations are omitted, no warning is produced. Instead the
+signature of the generated entry point method is equivalent to 
+``` c#
+    static void Main()
+```
+
+The example above would yield the following `$Main` method declaration:
+
+``` c#
+static class $Program
+{
+    static void $Main()
+    {
+        // Statements from File 1
+        if (args.Length == 0
+            || !int.TryParse(args[0], out int n)
+            || n < 0) return;
+        Console.WriteLine(Fib(n).curr);
+        
+        // Local functions from File 2
+        (int curr, int prev) Fib(int i)
+        {
+            if (i == 0) return (1, 0);
+            var (curr, prev) = Fib(i - 1);
+            return (curr + prev, curr);
+        }
+    }
+}
+```
+
+At the same time an example like this:
+``` c#
+// File 1
+await System.Threading.Tasks.Task.Delay(1000);
+System.Console.WriteLine("Hi!");
+```
+
+would  yield:
+``` c#
+static class $Program
+{
+    static async Task $Main()
+    {
+        // Statements from File 1
+        await System.Threading.Tasks.Task.Delay(1000);
+        System.Console.WriteLine("Hi!");
+    }
+}
+```
+
+### Scope of top-level local variables and local functions
+
+Even though top-level local variables and functions are "wrapped" 
+into the generated entry point method, they should still be in scope throughout the program.
```

Comment:
This line without the rest of the explanation suggests the wrong interpretation of the feature. They are not "in scope" as most users understand it (reachable and usable), they just "poison" it (reachable and unusable) to provide more useful diagnostics.